arch/sim/include/types.h

```diff
@@ -69,13 +69,41 @@ typedef unsigned char      _uint8_t;
 typedef signed short       _int16_t;
 typedef unsigned short     _uint16_t;
 
+#if defined(__INT32_TYPE__)
```

Comment:
should we apply the similar change to other arch? so we can decorate all printf/scanf like function with "format(...)" to catch the wrong type.

Reply:
i don't know. as i haven't checked other archs yet.

Reply:
i guess some of other 64-bit archs need it

Reply:
It isn't specific to 64bit arch, I think. Instead, it couple with toolchain(gcc or clang specific). It's easy to apply to other arch.

Reply:
i suppose that the specific issue i wanted to fix (ie. long long vs long mismatch) is only for 64-bit archs.
i agree it can be considered more general though.

Reply:
Yes, I think the consistent style and equally support all arch is NuttX's unique characteristics.